File: app/src/main/java/rusha/x/basket/Basket.java
package rusha.x.basket;

import com.google.gson.annotations.SerializedName;

import java.util.List;

public class Basket {
    @SerializedName("id")
    public final String id;

    @SerializedName("items")
    public final List<BasketItem> items;

    public Basket(String id, List<BasketItem> items) {
        this.id = id;
        this.items = items;
    }
}

File: app/src/main/java/rusha/x/basket/Basket.kt
package rusha.x.basket

import com.google.gson.annotations.SerializedName
import rusha.x.product.Product

data class Basket(
    @SerializedName("id") val id: String,
    @SerializedName("items") val items: List<Item>
) {
    data class Item(
        @SerializedName("count") val count: Double,
        @SerializedName("product") val product: Product
    )
}